utils/trivial-validate.py

```diff
@@ -42,6 +40,44 @@ def fail(s):
 with open(thispath + '/duplicate-whitelist.txt') as duplicate_fh:
     duplicate_allowed_list = [x.rstrip('\n') for x in duplicate_fh.readlines()]
 
+filenames = glob.glob('src/chrome/content/rules/*')
+
+def generate_hosts_count():
+    # Generate a count of the number of times a host appears, and on which
+    # platforms
+    hosts = {}
+    xml_parser = etree.XMLParser(remove_blank_text=True)
+
+    for filename in filenames:
+        if filename.endswith('/00README') or filename.endswith('/make-trivial-rule') or filename.endswith('/default.rulesets'):
+            continue
+        try:
+            tree = etree.parse(filename, xml_parser)
+        except Exception as oops:
+            print("%s failed XML validity: %s\n" % (filename, oops))
+            sys.exit(1)
+
+        try:
+            platform = xpath_ruleset(tree)[0].attrib["platform"]
```

Comment:
You can select this more directly with `/ruleset/@pattern`. And double-check what the results are when not present: you might be able to get away without the try/except.